src/mnemonic.rs

```diff
@@ -0,0 +1,41 @@
+/// FuelMnemonic is a simple mnemonic phrase generator.
+pub struct FuelMnemonic;
+
+#[cfg(feature = "std")]
+mod use_std {
+    use super::FuelMnemonic;
+    use crate::{Error, SecretKey};
+    use coins_bip32::path::DerivationPath;
+    use coins_bip39::{English, Mnemonic};
+    #[cfg(feature = "random")]
+    use rand::Rng;
+    use std::str::FromStr;
+
+    type W = English;
+
+    impl SecretKey {
```

Comment:
We shouldn't implement `SecretKey` stuff outside its module. Maybe create a submodule `mnemonic` and move this code there?

Reply:
Yup, that sounds good.